### PR TITLE
Searchbase

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
Integrate Input, props and Searchbase
Right now library size stands at `8.5kB`

image

### CODE REVIEW

File: src/components/VueSearchbox.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    onError: types.onError,
+    onResults: types.onResults,
+    innerClass: types.innerClass,
+    defaultQuery: types.defaultQuery,
+    beforeValueChange: types.beforeValueChange,
+    onQueryChange: types.onQueryChange,
+    className: types.className,
+    loader: types.loader,
+    onBlur: types.onBlur,
+    onKeyPress: types.onKeyPress,
```
Comment (jyash97) [inline src/components/VueSearchbox.jsx]: The event that we have on RS are like `keyPress` ,`blur`. 

The usage is different on vue :

```
<Searchbox @change="methodName" @keyPress="keyMethodName" />
```

Can you confirm this one if we support `onKeyPress` and it works properly?

File: src/components/VueSearchbox.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          fuzziness,
+          searchOperators
+        });
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  },
+  render() {
+    console.log(this.searchBase);
```
Comment (jyash97) [inline src/components/VueSearchbox.jsx:95]: Lets remove this

**Comment by jyash97** [comment]
Can you check bundle size by installing the dependency in an app and then use source map explorer?

You can do this locally by `npm link`

File: src/components/VueSearchbox.jsx (hunk 1 above)
Comment (ShahAnuj2610) [reply src/components/VueSearchbox.jsx]: handle in subsequent PR with `$emit`

File: src/components/VueSearchbox.jsx (hunk 2 above)
Comment (ShahAnuj2610) [reply src/components/VueSearchbox.jsx:95]: kept it deliberately for debugging, all `console.log` will be removed in the final PR

**Pull request merged**